src/Microsoft.Data.Analysis/Microsoft.Data.Analysis.csproj

```diff
@@ -0,0 +1,252 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+	<PropertyGroup>
+		<TargetFramework>netstandard2.0</TargetFramework>
+		<LangVersion>7.3</LangVersion>
+		<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+		<SuppressFinalPackageVersion>false</SuppressFinalPackageVersion>
+		<VersionPrefix>0.5.0</VersionPrefix>
+		<PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
+		<Description>This package contains easy-to-use and high-performance libraries for data analysis and transformation.</Description>
+		<PackageReleaseNotes>Initial preview of robust and extensible types and algorithms for manipulating structured data that supports aggregations, statistical funtions, sorting, grouping, joins, merges, handling missing values and more. </PackageReleaseNotes>
+		<PackageTags>ML.NET ML Machine Learning Data Science DataFrame Preparation DataView Analytics Exploration</PackageTags>
+		<GenerateDocumentationFile>true</GenerateDocumentationFile>
+		<!-- 
+      1591: Documentation warnings
+      NU5100: Warning that gets triggered because a .dll is not placed under lib folder on package. This is by design as we want MDAI to be under interactive-extensions folder.
+       -->
+		<NoWarn>$(NoWarn);1591;NU5100;MSML_GeneralName;MSML_ParameterLocalVarName;MSML_PrivateFieldName;MSML_TypeParamName;SA1028;SA1507;SX1101</NoWarn>
+		<TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);AddMDAIToInteractiveExtensionsFolder</TargetsForTfmSpecificContentInPackage>
+	</PropertyGroup>
+
+	<!-- The following properties are set to package M.D.A.Interactive with the M.D.A nuget package. If M.D.A.I undergoes TFM or dependency changes, we need to update the TargetFramework passed in below-->
+	<Target Name="AddMDAIToInteractiveExtensionsFolder">
+		<MSBuild Projects="./../Microsoft.Data.Analysis.Interactive/Microsoft.Data.Analysis.Interactive.csproj"
+				 Targets="_GetBuildOutputFilesWithTfm"
+				 Properties="TargetFramework=netcoreapp3.1">
+			<!-- Manually hardcoding the TargetFramework to netcoreapp3.1 as that is the one that MDAI targets -->
```

Comment:
This is pretty hacky.  @Anipik has been looking at defining patterns for dotnet/runtime pkgproj -> csproj transition.  I wonder if he has an idea for a better pattern for this case.

I see you're taking the one configuration of MS.D.A.Interactive, putting that assembly in `interactive-extensions/dotnet` and ignoring any of its dependencies.  Do you have any more info to share about how the `interactive-extensions/dotnet` folder works?  What is the expectation of components that use that convention?

Reply:
There isn't great documentation on this yet.

You can see some docs at https://github.com/dotnet/interactive/tree/main/docs#extending-net-interactive. Note there is an unlinked "Publishing your extension using NuGet" article that looks like it hasn't been written yet.

I think the best "doc" is the sample at:

https://github.com/dotnet/interactive/blob/main/samples/extensions/Library.nuget/Library.nuget.csproj

cc @jonsequitur @colombod @brettfo @LadyNaggaga

Reply:
What's the minimum TargetFramework required for dotnet interactive extensions?  Are dependencies allowed?  Should the extensions deps file be included? 

I think the way this is included now would add the file multiple times if M.D.A had multiple TargetFrameworks.  I think there might be a different hook other than `TfmSpecificPackageFile` to use for this scenario.  I'll see if I can come up with a sample.

Reply:
> What's the minimum TargetFramework required for dotnet interactive extensions? 

You need to reference: https://www.nuget.org/packages/Microsoft.DotNet.Interactive/1.0.0-beta.21155.3, which is a `netstandard2.1` library. So I guess that's the minimum? dotnet/interactive has already gone to `net5.0`, so I think we could bump this from `netcoreapp3.1` to `net5.0`.

> Are dependencies allowed?

I assume any dependencies you have outside of your NuGet package that aren't already dependencies of dotnet/interactive need to be included. But as it is now, `Microsoft.Data.Analysis.Interactive` doesn't have any external dependencies.

Reply:
I poked around the source and it looks like it just does a LoadFrom: https://github.com/dotnet/interactive/blob/6fd92eb9d72c119b6e7822ca87d96427399a3307/src/Microsoft.DotNet.Interactive/Extensions/AssemblyBasedExtensionLoader.cs#L85

I wonder how that plugin then finds the right copy of M.D.A 🤔   Anyhow, that's not what this comment was about.  That's a thought for another day.

Reply:
I was hacking around a bit with this and found that you could do this like the following which is marginally better:
```xml
    <BeforePack>$(BeforePack);IncludeInteractiveExtension</BeforePack>
  </PropertyGroup>

  <Target Name="IncludeInteractiveExtension">
    <MSBuild Projects="../ext/ext.csproj"
         Targets="GetTargetPath">
      <Output TaskParameter="TargetOutputs" ItemName="_interactiveExtension" />
    </MSBuild>
    <ItemGroup>
      <Content Include="@(_interactiveExtension)" Pack="True" PackagePath="interactive-extensions/dotnet" />
    </ItemGroup>
  </Target>
```

This works so long as the interactive extension doesn't cross-target, but it seems like that's a requirement already.  If you ever wanted to cross-target you could define a target in the extension project that ran in outer build and invoked the appropriate inner build framework to select the one that goes in the package (and avoid hardcoding the TFM in a different project).

Finally: what actually builds this project during pack?  I see that M.D.A can't take a project reference since M.D.A.I itself references M.D.A.  If you wanted the build to happen during pack you could change the target above to `Build` instead of `GetTargetPath`.